### ISSUE TITLE
Exhaustiveness for intersection types

### ISSUE THREAD

**Issue by scheglov** [dart-lang/sdk#51819]
Intuitively it feels that the switch in this code is exhaustive.
```dart
void f<T>(T x) {
  if (x is bool) {
    switch (x) {
      case true:
      case false:
        break;
    }
  }
}
```
But currently it is not.

**Comment by Pradeepyadav210** [comment]
yes but i can fix it. to make a switch exhaustive ,i can simply add one default value so the default value handle the other value other than
[solve issue.txt](https://github.com/dart-lang/sdk/files/11042937/solve.issue.txt)
 true or false statment
